### PR TITLE
applayer/htp: convert to new FAIL/PASS API v-3

### PR DESCRIPTION
**Pull request by Nancyenos**
Ticket: #6935

Make sure these boxes are checked accordingly before submitting your Pull Request -- thank you.

## Contribution style:
- [x] I have read the contributing guide lines at
   https://docs.suricata.io/en/latest/devguide/contributing/contribution-process.html

## Our Contribution agreements:
- [x] I have signed the Open Information Security Foundation contribution agreement at
   https://suricata.io/about/contribution-agreement/ (note: this is only required once)

## Changes (if applicable):
- [ ] I have updated the User Guide (in [doc/userguide/](https://github.com/OISF/suricata/tree/304271e63a9e388412f25f0f94a1a0da4bf619d9/doc/userguide)) to reflect the changes made
- [ ] I have updated the JSON schema (in [etc/schema.json](https://github.com/OISF/suricata/blob/304271e63a9e388412f25f0f94a1a0da4bf619d9/etc/schema.json)) to reflect all logging changes
      (including schema descriptions)
- [ ] I have created a ticket at
      https://redmine.openinfosecfoundation.org/projects/suricata/issues

Link to ticket: https://redmine.openinfosecfoundation.org/issues/

Describe changes:
-
- converted applayer/htp: to new FAIL/PASS API
-previous pr #11941 

### Provide values to any of the below to override the defaults.

- To use an LibHTP, Suricata-Verify or Suricata-Update pull request,
  link to the pull request in the respective `_BRANCH` variable.
- Leave unused overrides blank or remove.

SV_REPO=
SV_BRANCH=
SU_REPO=
SU_BRANCH=
LIBHTP_REPO=
LIBHTP_BRANCH=

### CODE REVIEW

File: src/app-layer-htp.c [hunk 1]
```diff
@@ -4084,147 +3923,88 @@ libhtp:\n\
...
-                goto end;
+                FAIL;
         }
         i++;
     }
 
-    ret = 1;
+    PASS;
 
-end:
```
Comment (victorjulien) [inline src/app-layer-htp.c:4227]: code from the old end block will be unreachable

**Review by victorjulien — CHANGES_REQUESTED** [review]
see inline

**Comment by inashivb** [comment]
Please also rebase your PR on top of the latest master branch.

**Comment by Nancyenos** [comment]
Than you for the review.working on it

**Pull request closed without merging**